org/postgresql/ssl/jdbc4/LibPQFactory.java

```diff
@@ -281,9 +283,45 @@ public boolean verify(String hostname, SSLSession session) {
         } else {
           return false;
         }
-      } else {
-        return CN.equals(hostname);
+      } else if(CN.equals(hostname.toLowerCase())){
+        return true;
       }
+      try {
+        for(List<?> i: serverCert.getSubjectAlternativeNames())
+        {
+          int type = (Integer) i.get(0);
+            switch(type)
+            {
+              case 2:
+                String dNSName = (String) i.get(1);
+                if(dNSName.toLowerCase().equals(hostname.toLowerCase()))
```

Comment:
Why don't you use `String#equalsIgnoreCase`? (open its implementation -- it does _both_ `toLower` and `toUpper`)